### PR TITLE
convert data download code to python pathlib

### PR DESCRIPTION
**Pull request by Mianto**
#369 @dstansby Please have a look over the code, Now it finally pass all test cases and I have rechecked the download files too.

### CODE REVIEW

File: heliopy/util/config.py [hunk 1]
```diff
@@ -49,18 +49,18 @@ def load_config():
...
     config_location = get_config_file()
     config = configparser.ConfigParser()
-    config.read(config_location)
+    config.read(str(config_location))
     config_dict = {}
 
     # Set data download directory
-    download_dir = os.path.expanduser(config['DEFAULT']['download_dir'])
+    download_dir = Path(config['DEFAULT']['download_dir']).expanduser()
     if os.name == 'nt':
```
Comment (abotiamnot) [inline heliopy/util/config.py]: You can remove this condition since you're using pathlib! Pathlib does it automatically.

Comment (abotiamnot) [inline heliopy/util/config.py]: You can remove this condition since you're using pathlib! Pathlib does it automatically.

Comment (abotiamnot) [reply heliopy/util/config.py]: Referring to the if os.name == 'nt'

**Comment by abotiamnot** [comment]
@Mianto There is another spice.py in the heliopy folder, right outside the data folder. There's an os.join statement conflicting with your build!

File: heliopy/data/util.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def process(dirs, fnames, extension, local_base_dir, remote_base_url,
         same length as *dirs*.
     extension : str
         File extension of the raw files. **Must include leading dot**.
-    local_base_dir : str
+    local_base_dir : PosixPath
```
Comment (dstansby) [inline heliopy/data/util.py]: Can this be a string or a `PosixPath`? Can then be converted to a Path using `local_base_dir = Path(local_base_dir)` inside the function.

File: heliopy/data/util.py [hunk 3]
```diff
@@ -360,9 +360,10 @@ def load(filename, local_dir, remote_url, guessversion=False,
 
     # Try to load locally
     if _checkdir(local_dir):
-        for f in os.listdir(local_dir):
-            if f == filename or guessversion and (f[:-6] == filename[:-6]):
-                filename = f
+        for f in local_dir.iterdir():
```
Comment (dstansby) [inline heliopy/data/util.py]: Does `local_dir.iterdir()` iterate through files in `local_dir`, or does it include the directory path too?

**Comment by Mianto** [comment]
@dstansby @abotiamnot I have gone through all the errors, all the errors finally connect to `util.load` but I am not able to find the exact bug. Can you please help me provide an idea where I am doing wrong?

**Comment by abotiamnot** [comment]
I've been failing the same tests your code is failing. It seems to do something with downloading the data.  

[Here's your test](https://travis-ci.org/heliopython/heliopy/jobs/350873312)
[Here's mine](https://travis-ci.org/abotiamnot/heliopy/jobs/351609244)

We are receiving the same errors ( TestImp::test_mag320ms passes on mine ). I've only modified Spice.py and it seems to be failing. 

Something to do with the server!

**Comment by dstansby** [comment]
Should be fixed now, I've restarted the travis tests to see what happens.

**Comment by dstansby** [comment]
Looks like there are still quite a few errors: https://travis-ci.org/heliopython/heliopy/jobs/362196363

Maybe doing this file by file would be easiest instead of all in one go?

**Comment by dstansby** [comment]
@Mianto do you have any plans to work on this in the future? No worries if not, but would be good to know so someone else can do it instead!

**Comment by Mianto** [comment]
@dstansby  I do plan to work on this issue. The errors are too much to tackle.

**Comment by dstansby** [comment]
I'm afraid I decided to make the changes in another PR, since we needed them to continue with the units project.

**Pull request closed without merging**